src/classes/RD2_EnablementDelegate_CTRL.cls

```diff
@@ -237,20 +239,56 @@ public with sharing class RD2_EnablementDelegate_CTRL extends STG_Panel {
         }
 
         try {
-            STG_Panel.stgService.stgRD.IsRecurringDonations2Enabled__c = true;
-
-            if (!Test.isRunningTest()) {
-                upsert STG_Panel.stgService.stgRD;
-                UTIL_OrgTelemetry_SVC.asyncProcessNpspSettingsTelemetry();
-            }
-
-            System.enqueueJob(new EnablementQueueable(QueueableMode.ScheduleJobs));
+            Boolean shouldRescheduleJobs = true;
+            enableEnhancement(shouldRescheduleJobs);
 
         } catch (Exception e) {
             throwAuraHandledException(e.getMessage());
         }
     }
 
+
+    /***
+    * @description Enables enhanced Recurring Donation feature from the callable API
+    * @return void
+    */
+    public static void callableApiEnableEnhancement(Boolean rescheduleBatchJobs) {
+        if (!STG_Panel.runningUserIsAdmin()) {
+            throw new UTIL_Permissions.InsufficientPermissionException(System.Label.RD2_EnablementPermissionRequired);
+        }
+
+        Boolean isCrlpEnabled = new CRLP_ApiService().isCrlpEnabled();
+        if (!isCrlpEnabled) {
+            throw new EnablementException(System.Label.CRLP_DisabledMessage);
+        }
+
+        if (service.isRecurringDonations2Enabled()) {
```

Comment:
I do not believe checking for RD2 already enabled is in the AC's. My preference is that if it's enabled already, it just does nothing.